### DISCUSSION TITLE
Apple device - How to start route for imported track (KML)

### DISCUSSION THREAD

**Discussion by Bab13** [organicmaps/organicmaps#2434] [Q&A] [answered]
Hi,
I've imported KML file and it is appearing in my favorites 💪
![image](https://user-images.githubusercontent.com/96130235/164439460-6e2f5cb9-ac28-414e-8654-c764d93a410e.png)

Now, I 'm wonderning how to start route for this track (appearing in red in below screenhot) ; I cannot found any button to start it ... Do this functionnality is existing ?

![image](https://user-images.githubusercontent.com/96130235/164439691-8b135200-d46e-4a17-9c69-072d8b0aa788.png)

Thanks a lot for your answer

Baptiste

**Comment by biodranik** [comment, marked as answer]
The route on the imported track is not implemented yet. How would you imagine it? What should happen if you are far from the track?

You can build a route with several intermediate points manually on top of it as a workaround.

**Reply by Bab13** [reply]
I was imagining that :
- if we are near the starting point of the trace, then start route would be triggered as it is currently possible when a route is created manually in the app
- if we are far from it, maybe, an option "Going to the starting point" would be be available in order to help me to go there

What do you think about it ? 

THanks

**Reply by biodranik** [reply]
That's a good start. Let's discuss other corner cases:
1. What if you are near the ending point or in the middle of the trace?
2. What if there is no map data under the track (e.g. there is a plain forest or empty space on the map under the track line)?

**Reply by Bab13** [reply]
1 In that case maybe 2 options could be assumed ; 
- either, app would propose to join nearest point of the trace
- or, app would propose to go to the starting point of the track

2 I assume it should be strange than a track is not under a map .... But, anyway, in that case, a message saying that "this route cannot be joined" can be raised. Neverthess, user can follow "manually" the track since the small arrow indicates the position

What do you think about it ?

**Reply by biodranik** [reply]
Good. What is the "manual" following the track?

**Reply by Bab13** [reply]
I mean that since in case 2, follow the route would not be possible, alternative for user is to double check that you are on the trace (thanks to basic GPS functionnality)

Is it helpful ?

**Reply by biodranik** [reply]
So "manual" following means "do not use any navigation mode, just show the map and position as before", right?

**Reply by Bab13** [reply]
Correct !

**Reply by biodranik** [reply]
Good, please follow #2384

**Reply by PureTryOut** [reply]
In my experience OSMand navigates to the nearest point of the track, which could be one in the middle, the very first one or even the last one. Just whichever one is closest. Personally I like it that way, but I can imagine an option to always route to the first point instead.
